README.md

```diff
@@ -28,9 +28,11 @@ The build process relies on environment variables specified in an `.env` file lo
 
 Make a copy of `.env.sample` and name it `.env`. You must make two modifications to this file before the docker containers will build.  Both `POSTGRES_PASSWORD` and `RABBIT_PASSWORD` should be uncommented and set with secure passwords.
 
+Before launching the site, ensure that the `PORT` variable is set correctly. For development, it should be set to `8000`; for deployment, it should be `80`. This variable not only sets the port the server listens on, it also configures the application's debug settings. Deploying the website with `PORT=8000` would leak debugging information on a live server and must be avoided.
```

Comment:
I'm not sure I understand this.

Port 80 is HTTP (443 is HTTPS). This is the default for a public site. Nginx, however, is your webserver that should be public-facing, and is thus proxying your Cantus application. 

Does this mean that you're running your docker on port 80? It doesn't need to (and, in fact, shouldn't -- ports < 1000 require admin privs, whereas ports > 1000 can run with their own (sandboxed) users.)

Reply:
> Nginx, however, is your webserver that should be public-facing, and is thus proxying your Cantus application.

Yes...although just a note here that the nginx server on Cantus Ultimus is not itself publicly accessible (or rather, it is accessible only from a small set of IP's: the DDMAL edge router and other DDMAL staging servers).

> Does this mean that you're running your docker on port 80? It doesn't need to (and, in fact, shouldn't -- ports < 1000 require admin privs, whereas ports > 1000 can run with their own (sandboxed) users.)

The nginx Docker container is listening on port 80, and this is either mapped to the host machine's port 80 (in the case of deployed versions, accepting incoming HTTP requests from the edge router) or the host machine's port 8000 (in the case of local development). The only reason it changes is because the django settings were configured (way back in the day) to depend on this environment variable to turn or off debug, the debug toolbar, etc. It would probably be a good idea to untangle these things, and certainly when I finally get to separating dev and prod dependencies!

In any case, I'm not sure I understand exactly what you are saying may not be correct? Does the above clarify? Raise new questions and concerns?

Reply:
I guess I'm more confused?

https://cantus.simssa.ca is on HTTPS, and the response says `Server: nginx`. This means that somewhere, some nginx is listening on 443. 

Are you saying that the path is `Internet -> nginx "edge router" (443) -> nginx (80) -> [some docker mapping] -> (gunicorn / django on 80? django on 8000?)`

Generally this would be set up as `Internet -> nginx (443) -> gunicorn (800N)`. If you have docker it might be something like `Internet -> nginx 443 -> docker 8000:8000 -> gunicorn 8000`. 

I guess what I'm saying is that if you have your edge router, you don't actually need (and, in fact, shouldn't really) be running on port 80 on your proxied services. 

Anyway, it obviously works for you, but it seems like a weird setup to me so I just thought I would clarify...

Reply:
> This means that somewhere, some nginx is listening on 443.

Yes, there is an nginx on the edge router that resolves the https and then passes an http request to the Cantus Ultimus server.

Path currently is: Internet -> nginx edge router (443) -> nginx CU  (80) -> gunicorn (8001)

Are you saying you would expect the staging edge router to proxy directly to gunicorn? 

Or just that you would expect it to proxy to nginx on the CU server but on a non-80 (eg. 8000) port?

Reply:
Yes. :-D

There's not really a need for the CU nginx, as far as i can see. The edge router is perfectly capable of proxying to gunicorn directly. Or if you want to keep the second nginx in the loop, then you can make it listen on 8000 (or any other port, really) since it doesn't really need to run on 80.

BTW, [This](https://github.com/DDMAL/cantus/blob/main/app/public/cantusdata/settings.py#L14-L15) seems like a really bizarre way of doing it, when you could just as easily define `os.environ.get("APP_DEPLOYMENT") == "DEVELOPMENT"` or something like that. But I wasn't there, so maybe there was a good reason for it?

It also looks like gunicorn is hardcoded to run on 8001. https://github.com/DDMAL/cantus/blob/main/app/django-config.sh#L8

Reply:
Gotcha. Ok, the only other thing that the CU-side nginx does is serve static files (e.g. Diva js + css, etc)...

> But I wasn't there, so maybe there was a good reason for it?

I kind of think this is one of those things where there have been a number of significant network changes over the life of CU (move to docker, move to using an edge router, etc) that this somewhat convoluted method is an artifact of that. 

I think what I'll do for the time being is remove the commits with this change and fix this in another PR.

Reply:
Ah, that could be the reason, if your static files are hosted on the upstream. Django isn't really great for serving static files, but nginx is.

Reply:
Will address the outstanding points here in #818